Doc/library/typing.rst

```diff
@@ -623,27 +623,30 @@ These can be used as types in annotations using ``[]``, each having a unique syn
 
 .. data:: Union
 
-   Union type; ``Union[X, Y]`` is equivalent to ``X | Y`` and means either X or Y.
+   Union type; ``X | Y`` is equivalent to ``Union[X, Y]`` and means either X or Y.
 
-   To define a union, use e.g. ``Union[int, str]`` or the shorthand ``int | str``.  Details:
+   To define a union, use e.g. ``int | str`` or the older version ``Union[int, str]``.
+   Note that the first version is preferred in readability in most cases.
+
+   Details:
 
    * The arguments must be types and there must be at least one.
 
    * Unions of unions are flattened, e.g.::
 
-       Union[Union[int, str], float] == Union[int, str, float]
+       Union[Union[int, str], float] == int | str | float
```

Comment:
I wonder if this might be confusing, we are using the two styles here for unions 🤔 

could we do something like:

```python
IntOrStr = int | str

IntOrStr | float == int | str | float
```

?

Reply:
Good idea. But, I guess we cannot completely remove `Union[Union[]]` example, because it is important as well. Maybe we can show both?

Reply:
why not :) hopefully it doesn’t get too verbose though